internal/activity/copy.go

```diff
@@ -0,0 +1,36 @@
+package activity
+
+import "io"
+
+// Copy copyes src to dst, pushing activity when bytes are read from src.
+func Copy(p *Pusher, dst io.Writer, src io.Reader) (written int64, err error) {
```

Comment:
A better way to represent this would be adding a `Writer(w io.Writer) io.Writer` method to pusher. Then, native io.Copy (or any transport) can be used.